### PR TITLE
Documentation should indicate how to verify binaries on binary download

### PR DESCRIPTION
**Pull request by dangoncalves**
* Specify which GPG key should be downloaded
 * Explain how to verify an official binary
 * Update documentation for macOS, GNU/Linux, and Microsoft Windows

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,38 +37,43 @@ brew install minio/stable/minio
...
 ## Microsoft Windows
 ### Binary Download
 | Platform| Architecture | URL|
 | ----------| -------- | ------|
 |Microsoft Windows|64-bit|https://dl.minio.io/server/minio/release/windows-amd64/minio.exe |
+|||https://dl.minio.io/server/minio/release/windows-amd64/minio.exe.asc |
 ```sh
-minio.exe server D:\Photos
+gpg2 --recv-keys 4405F3F0DDBA1B9E68A31D2512C74390F9AAC728
+gpg2 --verify minio.asc minio && minio.exe server D:\Photos
```
Comment (harshavardhana) [inline README.md:76]: minio.exe.asc is missing

File: README.md [hunk 2]
```diff
@@ -37,38 +37,43 @@ brew install minio/stable/minio
 | Platform| Architecture | URL|
 | ----------| -------- | ------|
 |Apple macOS|64-bit Intel|https://dl.minio.io/server/minio/release/darwin-amd64/minio |
+|||https://dl.minio.io/server/minio/release/darwin-amd64/minio.asc |
 ```sh
-chmod 755 minio
-./minio server /data
+gpg2 --recv-keys 4405F3F0DDBA1B9E68A31D2512C74390F9AAC728
+gpg2 --verify minio.asc minio && chmod 755 minio && ./minio server /data
```
Comment (harshavardhana) [inline README.md:43]: Write each command explicitly in separate lines, it becomes cryptic to read.

File: README.md [hunk 3]
```diff
@@ -37,38 +37,43 @@ brew install minio/stable/minio
...
 | ----------| -------- | ------|
 |GNU/Linux|64-bit Intel|https://dl.minio.io/server/minio/release/linux-amd64/minio |
+|||https://dl.minio.io/server/minio/release/linux-amd64/minio.asc |
 ```sh
-wget https://dl.minio.io/server/minio/release/linux-amd64/minio
-chmod +x minio
-./minio server /data
+wget https://dl.minio.io/server/minio/release/linux-amd64/minio{,.asc}
+gpg2 --recv-keys 4405F3F0DDBA1B9E68A31D2512C74390F9AAC728
+gpg2 --verify minio.asc minio && chmod 755 minio && ./minio server /data
```
Comment (harshavardhana) [inline README.md:55]: Write each command explicitly in separate lines, it becomes cryptic to read.

File: README.md [hunk 4]
```diff
@@ -37,38 +37,43 @@ brew install minio/stable/minio
...
 | ----------| -------- | ------|
 |GNU/Linux|ppc64le|https://dl.minio.io/server/minio/release/linux-ppc64le/minio |
+|||https://dl.minio.io/server/minio/release/linux-ppc64le/minio.asc |
 ```sh
-wget https://dl.minio.io/server/minio/release/linux-ppc64le/minio
-chmod +x minio
-./minio server /data
+wget https://dl.minio.io/server/minio/release/linux-ppc64le/minio{,.asc}
+gpg2 --recv-keys 4405F3F0DDBA1B9E68A31D2512C74390F9AAC728
+gpg2 --verify minio.asc minio && chmod 755 minio && ./minio server /data
```
Comment (harshavardhana) [inline README.md:65]: Write each command explicitly in separate lines, it becomes cryptic to read.

File: README.md [hunk 5]
```diff
@@ -37,38 +37,43 @@ brew install minio/stable/minio
...
 
 | Platform| Architecture | URL|
 | ----------| -------- | ------|
 |GNU/Linux|ppc64le|https://dl.minio.io/server/minio/release/linux-ppc64le/minio |
+|||https://dl.minio.io/server/minio/release/linux-ppc64le/minio.asc |
 ```sh
-wget https://dl.minio.io/server/minio/release/linux-ppc64le/minio
-chmod +x minio
-./minio server /data
+wget https://dl.minio.io/server/minio/release/linux-ppc64le/minio{,.asc}
```
Comment (harshavardhana) [inline README.md:63]: What is the need for , ?? . We don't create any file other than minio.asc

File: README.md (hunk 1 above)
Comment (harshavardhana) [reply README.md:76]: Windows binaries are already signed using Windows cert signature, we don't need to talk about gpg verification.

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Thanks for the PR some comments.

File: README.md (hunk 5 above)
Comment (dangoncalves) [reply README.md:63]: In order to verify minio, we need to download the minio binary itself and the signature too. This command will download minio and minio.asc in the same command. This is known as brace expansion (for more info, refer to [the bash manual](https://www.gnu.org/software/bash/manual/html_node/Brace-Expansion.html)).

Do you prefer this form (a little bit cryptic but we can add comment), or should we execute two wget commands?

Comment (harshavardhana) [reply README.md:63]: Yes I know but let's not add such cryptic expansions, copy pasting with such things is prone to errors and we would like to keep it simple.

We should also mention verification separately, there is also sha256sum which should be mentioned.

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Now that I think about it we already have a comprehensive documentation on this https://github.com/minio/cookbook/blob/master/docs/verifying-minio-binaries.md 

We can simply reference this appropriately here. Simplicity of current command line docs shouldn't be disturbed, and provides a more comprehensive support for checksums as well.

**Comment by deekoder** [comment]
@dangoncalves - Are you able to discuss further changes to this pr?

**Comment by deekoder** [comment]
Closing this pr as stale. @dangoncalves  please let's continue the discussion on this when you are available.

**Pull request closed without merging**